src/neuroconv/datainterfaces/ecephys/ced/ceddatainterface.py

```diff
@@ -9,12 +9,20 @@ def _test_sonpy_installation() -> None:
     get_package(package_name="sonpy", excluded_platforms_and_python_versions=dict(darwin=["3.7"]))
 
 
-class CEDRecordingInterface(BaseRecordingExtractorInterface):
-    """Primary data interface class for converting data from CED (Cambridge Electronic Design)."""
+class LazyExtractorClass(type(BaseRecordingExtractorInterface), type):
+    def __getattribute__(self, name):
+        if name == "RX":
+            _test_sonpy_installation()
+            spikeinterface = get_package(package_name="spikeinterface")
+            RX = spikeinterface.extractors.CedRecordingExtractor
+            self.RX = RX
+
+            return RX
+        return super().__getattribute__(name)
```

Comment:
Also a bit touchy - this must be defined this way in order for decorators like `classmethod` to become bound to the corresponding functions like `get_source_schema`.